src/adapters/lokijs/worker/executor.js

```diff
@@ -257,7 +257,12 @@ export default class LokiExecutor {
 
       if (migrationSteps) {
         logger.log(`[DB][Worker] Migrating from version ${dbVersion} to ${this.schema.version}...`)
-        await this._migrate(migrationSteps)
+        try {
+          await this._migrate(migrationSteps)
+        } catch (error) {
+          logger.error('[DB][Worker] Migration failed. Resetting database instead', error)
```

Comment:
Woah, woah, woah, that's a little bit radical don't you think? What if someone is using Watermelon as an offline database without syncing with server and migration somehow errored? User is going to lose all of his data without a way to rollback. Frankly I don't really know if this is standard practice  in database migrations or not.

Reply:
I think @radex is right and this should be default behavior. Maybe we could add a flag if someone wants to use Watermelon as offline-only.

The only downside is that we don't provide sync engine, so Watermelon is not going to be usable (reliably) on its own :/